File: dotnet-api/Models/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace dotnet_api.Models
{
    public class Item
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public string Category { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public string ImageUrl { get; set; } = string.Empty;
        public ulong Factorial { get; set; }
    }
}

File: angular-ui/src/app/items/item.ts
export interface Item {
    id: number;
    name: string;
    description: string;
    price: number;
    quantity: number;
    category: string;
    createdAt: string;
    imageUrl: string;
    factorial: number;
}

export interface CreateItemDto {
    name: string;
    description: string;
    price: number;
    quantity: number;
    category: string;
    imageUrl: string;
}